game.py

```diff
@@ -14,13 +14,67 @@
 
 
 def snowman(snowman_word):
-    """Complete the snowman function
-    replace "pass" below with your own code
-    It should print 'Congratulations, you win!'
-    If the player wins and, 
-    'Sorry, you lose! The word was {snowman_word}' if the player loses
-    """
-    pass
+
+    # print(snowman_word)
+
+    # creates a constant that stores the length of the snowman word
+    WORD_LENGTH = len(snowman_word) 
+
+    # using the snowman word, creates a dictionary that stores the letters and their current guesssed status
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    # list that stores the guessed incorrect letters
+    wrong_guesses_list = []
+    # boolean that stores the puzzle win state
+    puzzle_solved = False
+
+    # while loop with the following exit conditions:
+    # - the length of wrong_guesses_list equals or exceeds SNOWMAN_MAX_WRONG_GUESSES
+    # - the puzzle_solved boolean's state is changed to True
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and puzzle_solved is not True:
+
+        # print statement which calls the function that displays the current state of the snowman word
+        print(generate_word_progress_string(snowman_word, correct_letter_guess_statuses))
```

Comment:
We could also use the [provided helper function](https://github.com/Ada-C23/snowman/blob/main/game.py#L74C5-L82) `print_word_progress_string` to print this out